### PR TITLE
Add python unittests

### PR DESCRIPTION
**Pull request by BPerlakiH**
Fixes: #48 
Run python unit tests as part of CI.

### CODE REVIEW

**Comment by BPerlakiH** [comment]
The working CI steps can be checked here: https://github.com/kiwix/kiwix-apple-custom/actions/runs/7898577866/job/21556249437

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,11 +30,20 @@ jobs:
...
         uses: actions/checkout@v4
         with:
-          path: custom
+          path: custom 
+
+      - name: Install Python dependencies for custom project generation
+        run: python -m pip install pyyaml
+      
+      - name: Test python scripts
+        run: |
```
Comment (rgaudin) [inline .github/workflows/ci.yml:39]: You don't need to cd back as steps are independent.

You can even simplify with

```yaml
    name: xxx
    working-directory: custom
    env:
      PYTHONPATH: src/
    run: python -m unittest
```

The PYTHONPATH is so you don't fumble with paths in source code

File: tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+import sys
```
Comment (rgaudin) [inline tests/__init__.py:1]: Don't do that in code (as suggested above).

**Review by rgaudin — CHANGES_REQUESTED** [review]

File: tests/__init__.py (hunk 2 above)
Comment (BPerlakiH) [reply tests/__init__.py]: Thank you, now removed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (BPerlakiH) [reply .github/workflows/ci.yml]: Thank you, It's good to know. I've updated it.

**Review by rgaudin — APPROVED** [review]
👍

**Pull request merged**